### ISSUE TITLE
A small bug in ModularityOptimizer

### ISSUE THREAD

**Issue by LittleBearX** [satijalab/seurat#2451]
@evolvedmicrobe 
Hello:
I am reading the Source code of Louvain algorithm in src/ModularityOptimizer.cpp.  Thanks for your translation from java. I am trying to rewrite it by python. But I have a problem, is there a problem in **line 247**?     
```cpp
std::fill(edge[0].begin() + firstNeighborIndex.at(i), edge[0].end() + firstNeighborIndex.at(i + 1), i);
```
I think it should be:
```cpp
std::fill(edge[0].begin() + firstNeighborIndex.at(i), edge[0].begin() + firstNeighborIndex.at(i + 1), i);
```
Thanks!

**Comment by evolvedmicrobe** [comment]
@LittleBearX nice catch! Indeed, that should have been `begin/begin` instead of `begin/end` like in the other calls, as in it's current form it would likely cause the `std::fill` method to write past the end of the array and perhaps give a SIGSEGV.  It looks like that method isn't actually called anywhere in the code though, which is likely why it's been missed until now.  I gather you found this while calling the method in a different way?

I'll submit a PR with the fix!

Thanks!
Nigel

**Comment by andrewwbutler** [comment]
Thanks for the report and fix (#2469). This is now updated on the `develop` branch.
